docs/spire.md

```diff
@@ -0,0 +1,273 @@
+1. The SPIRE agent will attest the workload and request an SVID.
+1. The entrypointer receives an x509 SVID, containing the x509 certificate and associated private key. 
+1. The entrypointer signs the results of the TaskRun and emits the signatures and x509 certificate to the TaskRun results for later verification.
+
+## Enabling TaskRun result attestations
+
+To enable TaskRun attestations:
+1. Make sure `enforce-nonfalsifiability` is set to `"spire"` in the `feature-flags` configmap, see [`install.md`](./install.md#customizing-the-pipelines-controller-behavior) for details
+1. Create a SPIRE deployment containing a SPIRE server, SPIRE agents and the SPIRE CSI driver, for convenience, [this sample single cluster deployment](https://github.com/spiffe/spiffe-csi/tree/main/example/config) can be used.
+1. Register the SPIRE workload entry for Tekton with the "Admin" flag, which will allow the Tekton controller to communicate with the SPIRE server to manage the TaskRun identities dynamically.
+    ```
+
+    # This example is assuming use of the above SPIRE deployment
+    # Example where trust domain is "example.org" and cluster name is "example-cluster"
+    
+    # Register a node alias for all nodes of which the Tekton Controller may reside
+    kubectl -n spire exec -it \
+        deployment/spire-server -- \
+        /opt/spire/bin/spire-server entry create \
+            -node \
+            -spiffeID spiffe://example.org/allnodes \
+            -selector k8s_psat:cluster:example-cluster
+    
+    # Register the tekton controller workload to have access to creating entries in the SPIRE server
+    kubectl -n spire exec -it \
+        deployment/spire-server -- \
+        /opt/spire/bin/spire-server entry create \
+            -admin \
+            -spiffeID spiffe://example.org/tekton/controller \
+            -parentID spiffe://example.org/allnode \
+            -selector k8s:ns:tekton-pipelines \
+            -selector k8s:pod-label:app:tekton-pipelines-controller \
+            -selector k8s:sa:tekton-pipelines-controller
+    
+    ```
+    
+1. Modify the controller (`config/controller.yaml`) to provide access to the SPIRE agent socket.
+    ```yaml
+    # Add the following the volumeMounts of the "tekton-pipelines-controller" container
+    - name: spiffe-workload-api
+      mountPath: /spiffe-workload-api
+      readOnly: true
+    
+    # Add the following to the volumes of the controller pod
+    - name: spiffe-workload-api
+      csi:
+        driver: "csi.spiffe.io"
+    ```
+1. (Optional) Modify the configmap (`config/config-spire.yaml`) to configure non-default SPIRE options.
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: config-spire
+      namespace: tekton-pipelines
+    labels:
+      app.kubernetes.io/instance: default
+      app.kubernetes.io/part-of: tekton-pipelines
+    data:
+      # spire-trust-domain specifies the SPIRE trust domain to use.
```

Comment:
is it possible to provide some more details around how to know what values to provide here? e.g. links to spire docs maybe? (i know i wouldn't know what value to put for 'trust domain' if i was trying to set this up 😅 )

Reply:
Yes that is a good suggestion. Here is a link to it: https://spiffe.io/docs/latest/deploying/spire_server/#server-configuration-file